imports/plugins/core/orders/client/components/OrderCardFulfillmentGroupStatusButton.js

```diff
@@ -0,0 +1,147 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { Mutation } from "react-apollo";
+import withStyles from "@material-ui/core/styles/withStyles";
+import FormControl from "@material-ui/core/FormControl";
+import Grid from "@material-ui/core/Grid";
+import InputLabel from "@material-ui/core/InputLabel";
+import MenuItem from "@material-ui/core/MenuItem";
+import OutlinedInput from "@material-ui/core/OutlinedInput";
+import Select from "@material-ui/core/Select";
+import { i18next, Reaction } from "/client/api";
+import ConfirmButton from "/imports/client/ui/components/ConfirmButton";
+import updateOrderFulfillmentGroupMutation from "../graphql/mutations/updateOrderFulfillmentGroup";
+
+const styles = (theme) => ({
+  formControl: {
+    margin: theme.spacing.unit,
+    minWidth: 120
+  }
+});
+
+class OrderCardFulfillmentGroupStatusButton extends Component {
+  static propTypes = {
+    classes: PropTypes.object,
+    fulfillmentGroup: PropTypes.shape({
+      _id: PropTypes.string,
+      items: PropTypes.object,
+      selectedFulfillmentOption: PropTypes.shape({
+        fulfillmentMethod: PropTypes.shape({
+          carrier: PropTypes.string
+        })
+      }),
+      status: PropTypes.string
+    })
+  };
+
+  state = {
+    labelWidth: 100,
+    status: this.props.fulfillmentGroup.status
+  };
+
+  handleSelectChange = (event) => {
+    this.setState({
+      [event.target.name]: event.target.value
+    });
+  }
+
+  async handleUpdateFulfillmentGroupStatus(mutation) {
+    const hasPermission = Reaction.hasPermission(["reaction-orders", "order/fulfillment"], Reaction.getUserId(), Reaction.getShopId());
+    const { fulfillmentGroup, order } = this.props;
+
+    if (hasPermission) {
+      await mutation({
+        variables: {
+          orderFulfillmentGroupId: fulfillmentGroup._id,
+          orderId: order._id,
+          status: this.state.status
+        }
+      });
+    }
+  }
+
+  // TODO: When MUI 4.x is implemented, change this to `SplitButton`
+  // instead of using a select dropdown in a popup
+  // https://material-ui.com/components/buttons/#split-button
+  render() {
+    const hasPermission = Reaction.hasPermission(["reaction-orders", "order/fulfillment"], Reaction.getUserId(), Reaction.getShopId());
+    const { classes, fulfillmentGroup } = this.props;
+    const canUpdateFulfillmentStatus = (fulfillmentGroup.status !== "coreOrderWorkflow/canceled");
+    const options = [
```

Comment:
Are these options used anywhere else? or will they be used in the future in another component? if so, they should like in a `orderStatuses` helper.

Reply:
Currently, no, because we want to restructure all these workflows in the near future anyway, so this is the only place they are used.